components/src/histogram.tsx

```diff
@@ -19,18 +19,27 @@
  *
  */
 
-import iobioCharts from 'iobio-charts';
-import { useEffect } from 'react';
+import 'iobio-charts';
 
-const { createHistogram } = iobioCharts;
-
-function Histogram(props: { 'broker-key': string }) {
-	useEffect(() => {
-		createHistogram();
-	}, []);
-
-	// return <iobio-histogram />;
-	return <div broker-key={props['broker-key']}></div>;
+function IobioHistogram({
+	brokerKey,
+	color,
+	ignoreOutliers,
+	title,
+}: {
+	brokerKey: string;
+	color?: string;
+	ignoreOutliers?: boolean;
+	title?: string;
+}) {
+	return (
+		<iobio-histogram
+			broker-key={brokerKey}
+			ignore-outliers={ignoreOutliers}
+			title={title}
+			style={{ '--iobio-data-color': color }}
```

Comment:
Color is a good one to test lifecycle methods / prop updates, but TBH I'm not sure how necessary they are -- i.e. why would broker-key change?
could cross that bridge when we need to

Reply:
The `props` don't necessarily have to change to be useful. `state` has to change to be useful.

These components need to be reusable. I sometimes think of a `prop` like a component config when it doesn't need to change. 

Example: two different projects are likely to have different `brokerKey` values. `props` allows the consumer of the component to provide that information. it's configurable instead of hardcoding each var into the code.

Example: We have two histograms with different titles. Having a `title` prop facilities reusability without having to write two distinct components.

Reply:
I'm only questioning the need for lifecycle methods
The props are all more akin to config than state that will change and update

Reply:
The prop is scoped to `color` and uses `style` so it's just scoped to this component, and the CSS variable was the Iobio team's design.

I'm less concerned about the color variable leaking; I think it's a good question what we should do if we want to style all `title` elements for example.

Reply:
The main need for lifecycle methods is using the web components library in a "React way".
The components are functional right now because there's a lot of heavy lifting in them 
1) define the components as HTML custom elements. and
2) pop the data in them from the data broker
So at this point React is just rendering them happy as Larry because it doesn't really need to worry about anything. Just like a `div` or a simply element as far as React cares.

Moving forwards....
What if our React components are doing data handling? Then holding off rendering the chart until something `mounts` might come into play. What if we need to "disconnect" from a data source when the Histogram isn't in view any more? Cleanup code.. looking at `useEffect` cleanup methods. 

Anything that the iobio library does that React doesn't know about, you need to tell it.
For instance on a JQuery library integration with React, if you manipulate the DOM with JQuery which is very common, React doesn't magically pickup the changes, you have to tell it to render again. Which is the need for integration.


I think it'll be more clear where the lifecycle methods are useful the more development happens. 
I don't know about this project scope or roadmap so not sure the scale of this integration.

Does that help at all?

Reply:
`I think it'll be more clear where the lifecycle methods are useful the more development happens`
This is exactly what I was trying to express originally -- that lifecycle methods may not be needed at this stage

But this conversation gave me an idea and I found there are some side effects when you Show/Hide the entire BAM section. Hiding the Data Broker causes errors and you need some sort of handler to gracefully mount/unmount everything.

Reply:
Revisiting styles -- I have a draft version using constructable style sheets coming shortly
https://webcomponents.guide/learn/components/styling/#using-constructable-stylesheets

Reply:
Finally got this working
I use a CSS string (because that's what `CSSStyleSheet.replaceSync()` expects) so if style objects are preferred then we'll need a conversion tool
But I've added examples for styling specific elements (.iobio-histogram-title) as well as updating the color variables

https://github.com/overture-stack/iobio-components/pull/24#:~:text=2742a5d-,Custom%20Styles%20Updated,-439ef5d

Reply:
I see, before it was a standard `style` tag being passed through to the element. CSS var essentially being global, so it was affecting colour.

This changes it so we're passing `styles` which does not have special meaning. Then adding the CSS string styling to the components style.

Checking this out now. An odd thing is this doesn't appear to live update on code change.